tests/Common/BranchAwareClientTest.php

```diff
@@ -31,19 +34,30 @@ public function setUp()
         }
     }
 
+    /**
+     * @dataProvider provideDefaultBranchUsage
+     * @return void
+     */
     public function testClientWithDefaultBranch()
     {
-        $devBranch = new DevBranches($this->_client);
-
-        $defaultBranches = array_filter(
-            $devBranch->listBranches(),
-            function (array $branch) {
-                return $branch['isDefault'] === true;
-            }
-        );
-
-        $defaultBranch = reset($defaultBranches);
-        $branchClient = $this->getBranchAwareDefaultClient($defaultBranch['id']);
+        switch ($this->dataName()) {
```

Comment:
Néni to docela zkryplené to dělat před `$this->dataName()`
Proč ten provider nevracá třeba, a nepoužívás normálně argumenty?
```
        return [
            'use default branch by id' => ['id',null],
            'use default branch by placeholder' => ['placeholder','default'],
        ];
```

Reply:
protoze jak jsem to mel, tak mi prislo blby mit `'id' => ['id'], 'placeholder' => ['placeholder']`... 🤷 
Takhle jak to mas ty, mi to prijde uz OK. Upravim.